File: src/etch/bindings/java/transport/fmt/binary/BinaryTaggedDataFormatFactory.java
package etch.bindings.java.transport.fmt.binary;

import etch.bindings.java.msg.ValueFactory;
import etch.bindings.java.transport.FormatFactory;
import etch.bindings.java.transport.TaggedDataInput;
import etch.bindings.java.transport.TaggedDataOutput;

/**
 * Format factory for binary tagged data.
 */
public class BinaryTaggedDataFormatFactory extends FormatFactory
{
	@Override
	public TaggedDataInput newTaggedDataInput( ValueFactory vf, String uri )
	{
		return new BinaryTaggedDataInput( vf, uri );
	}

	@Override
	public TaggedDataOutput newTaggedDataOutput( ValueFactory vf, String uri )
	{
		return new BinaryTaggedDataOutput( vf, uri );
	}
}

File: src/etch/bindings/csharp/transport/fmt/binary/BinaryTaggedDataFormatFactory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Etch.Transport.Fmt.Binary
 
{
    public class BinaryTaggedDataFormatFactory : FormatFactory
    {
        
        public BinaryTaggedDataFormatFactory() {}
        
        public override TaggedDataInput NewTaggedDataInput( Etch.Msg.ValueFactory vf, string uri )
        {
            return new Etch.Transport.Fmt.Binary.BinaryTaggedDataInput( vf, uri );
        }

        public override TaggedDataOutput NewTaggedDataOutput( Etch.Msg.ValueFactory vf, string uri )
        {
            return new Etch.Transport.Fmt.Binary.BinaryTaggedDataOutput( vf, uri );
        }
    }
}
